common/setups/rasr/gmm_system.py

```diff
@@ -826,6 +830,46 @@ def sat_recognition(
                     **kwargs,
                 )
 
+    # -------------------- output helpers  --------------------
+
+    def get_gmm_output(
+        self, corpus_key: str, corpus_type: str, step_idx: int, steps: RasrSteps
+    ):
+        """
+        :param corpus_key: corpus name identifier
+        :param corpus_type: corpus used for: train, dev or test
+        :param step: step name in the gmm pipeline: tri, vtln, ...
+        :param iteration: which iteration to take within one step
+        :return GmmOutput:
+        """
+        gmm_output = GmmOutput()
+        gmm_output.crp = self.crp[corpus_key]
+        gmm_output.corpus_file = self.corpora[corpus_key].corpus_file
+        gmm_output.corpus_object = self.corpora[corpus_key]
+        gmm_output.lexicon_file = self.crp[corpus_key].lexicon_config.file
+        gmm_output.lexicon = self.crp[corpus_key].lexicon_config
+        gmm_output.cart_tree = self.crp[
+            corpus_key
+        ].acoustic_model_config.state_tying.file
+        gmm_output.feature_flows = self.feature_flows[corpus_key]
+        gmm_output.features = self.feature_caches[corpus_key]
+        gmm_output.segment_path = self.crp[corpus_key].segment_path
+        gmm_output.allophone_file = self.allophone_files["base"]
+
+        if corpus_type == "dev" or corpus_type == "test":
+            gmm_output.lm_file = self.crp[corpus_key].language_model_config.file
+            gmm_output.lm = self.crp[corpus_key].language_model_config
+            gmm_output.acoustic_mixtures = self.mixtures[corpus_key][
+                f"train_{steps.get_step_names_as_list()[step_idx - 1]}"
+            ][-1]
+
+        if corpus_type == "train":
+            gmm_output.alignments = self.alignments[corpus_key][
+                f"train_{steps.get_step_names_as_list()[step_idx - 1]}"
+            ][-1]
```

Comment:
Also dev sets can have alignments, e.g. if we use them for cross validation of a ce loss.

Reply:
I agree. I ll add this